turbopack/crates/turbopack-core/src/code_builder.rs

```diff
@@ -4,24 +4,18 @@ use std::{
     ops,
 };
 
-use anyhow::{Context, Result};
-use indexmap::IndexMap;
-use turbo_tasks::{ResolvedVc, Vc};
-use turbo_tasks_fs::{
-    rope::{Rope, RopeBuilder},
-    util::uri_from_file,
-    DiskFileSystem, FileSystemPath,
-};
+use anyhow::Result;
+use turbo_tasks::Vc;
+use turbo_tasks_fs::rope::{Rope, RopeBuilder};
 use turbo_tasks_hash::hash_xxh3_hash64;
 
 use crate::{
-    source_map::{GenerateSourceMap, OptionSourceMap, SourceMap, SourceMapSection},
+    source_map::{GenerateSourceMap, OptionStringifiedSourceMap, SourceMap},
     source_pos::SourcePos,
-    SOURCE_MAP_PREFIX,
 };
 
 /// A mapping of byte-offset in the code string to an associated source map.
-pub type Mapping = (usize, Option<ResolvedVc<Box<dyn GenerateSourceMap>>>);
+pub type Mapping = (usize, Option<Rope>);
```

Comment:
```suggestion
pub type Mapping = (usize, Option<ResolvedVc<Rope>>);
```

It would be better if we keep these Ropes in Vcs for better persistent caching. Otherwise it duplicates the data when serialized